internal/pkg/agent/application/coordinator/testdata/simple_config/expected/components-expected.yaml

```diff
@@ -422,6 +422,10 @@ components:
     id: log-default
     log_level: 2
     type: 1
+  features:
+    fqdn:
+      enabled: false
```

Comment:
@pchila can correct me if I'm wrong here but I believe that's because the `state.yaml` that's generated when diagnostics are requested is being generated from components' states, which can be set in the test's code.  I'm doing that in this PR over here:

https://github.com/elastic/elastic-agent/pull/2218/files#diff-fc6b455c3e3a0055095709cc4138f976720bf752acce1ac872505bcf94bf9322R158-R161.

By comparison, the other files that are generated diagnostics are requested, e.g. the `components-expected.yml` here, tha are being generated from the `elastic-agent.yml` test fixture.  I didn't touch that one in this PR so it has [`features` set to an empty object](https://github.com/elastic/elastic-agent/blob/fd41fe4e239474d9ae001ad7daf6b99cb398e70c/internal/pkg/agent/application/coordinator/testdata/simple_config/elastic-agent.yml#L156).

Reply:
The general idea of this test is to verify that given a certain policy (`elastic-agent.yaml` part) and runtime state (tried to put that in as a yaml as well) we get correctly reported diagnostics.

As you correctly said, the runtime part is mocked away so a testcase needs to set up a given state to be returned by the mock: I tried doing that using another yaml input in `components-units.yaml` but I had limited success in creating the necessary structs from yaml (look at https://github.com/elastic/elastic-agent/pull/2380#issuecomment-1484791151 for more details) 

For the existing testcase: it was meant to test a basic policy we can create on fleet on verify that the diagnostics are ok assuming all units/components healthy and a linux system.
In such a usecase I would expect the `features` part to be rendered with their default values when there is no input specified.

If you need to test the features being set in policy I would suggest to create a separate usecase (by adding a separate directory under `testdata` with its own inputs and expected outputs) but then we would need to fix the way the components and units states are mocked.

Conversely, if the default case for features needs an adjustment of the existing usecase we can modify the existing code/inputs  to fit.

To summarize: I would suggest we get a correct test for the existing default policy input (that means that state and config should be coherent in that testcase with a default policy we can get from fleet) and, if needed, add a separate usecase for features (can be simpler than the default policy)

Reply:
Thank for the detailed explanation and recommendation, @pchila.   I'm thinking maybe I could split the difference - between writing a new test case and testing the default feature flags setup, where they are all disabled - and enable the FQDN feature flag in the test's `elastic-agent.yml`.   And, based on your explanation, it sounds like I would mock the runtime state for feature flags to match this test scenario. 

Do you see any issues with this approach?

Reply:
Made the change in my previous comment in df05966c8d9106e6272d8668e84cac13f8ce8b72.  Hope it makes sense, @pchila.